compiler/rustc_middle/src/mir/patch.rs

```diff
@@ -93,20 +98,20 @@ impl<'tcx> MirPatch<'tcx> {
         bb
     }
 
-    pub fn terminate_block(&mut self) -> BasicBlock {
-        if let Some(bb) = self.terminate_block {
-            return bb;
+    pub fn terminate_block(&mut self, reason: UnwindTerminateReason) -> BasicBlock {
+        if let Some((cached_bb, cached_reason)) = self.terminate_block && reason == cached_reason {
```

Comment:
This can go wrong in a nounwind function because both `UnwindTerminate` reasons would exists and many new blocks will be generated.

Reply:
Well it wouldn't go wrong in the sense of doing anything wrong. But yeah the cache wouldn't be as effective any more. Is that case common enough to be worth optimizing for? My conclusion was, no.